core/src/main/java/ai/timefold/solver/core/impl/domain/entity/descriptor/EntityDescriptor.java

```diff
@@ -352,7 +366,14 @@ The entityClass (%s) has a @%s annotated member (%s) that has an unsupported typ
             declaredShadowVariableDescriptorMap.put(memberName, variableDescriptor);
         } else if (variableAnnotationClass.equals(ShadowVariable.class)
                 || variableAnnotationClass.equals(ShadowVariable.List.class)) {
-            var variableDescriptor = new CustomShadowVariableDescriptor<>(nextVariableDescriptorOrdinal, this, memberAccessor);
+            ShadowVariableDescriptor<Solution_> variableDescriptor;
+            var annotation = memberAccessor.getAnnotation(ShadowVariable.class);
+            if (annotation != null && !annotation.method().isEmpty()) {
+                variableDescriptor =
```

Comment:
Where are the fail-fasts for when the other fields are used? The Javadoc suggests that they exist, but I didn't notice them.

Reply:
Fail fasts are in RootVariableSource, which does the processing of the variable paths.